### ISSUE TITLE
Emit should not failed while reconnecting

### ISSUE THREAD

**Issue by SSebo** [1c3t3a/rust-socketio#230]
like js client, should buffer the emit, resend if reconnect success.

**Comment by TomieAi** [comment]
how you reconnect?

**Comment by 1c3t3a** [comment]
This is closed in #229 isn't it?

**Comment by SSebo** [comment]
No, `emit` still return `IllegalActionBeforeOpen` Error. Maybe we should cache the payload when reconnecting, can emit after reconnect.
